File: capacity-club-api/src/domain-modules/indoor-training-subscription/model/payload/indoor-training-subscription-update.payload.ts
import { UniqueId } from '@common/model';
import { Payload } from '@domain-modules-shared';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsOptional, Length } from 'class-validator';
import { IndoorTraining } from 'domain-modules/indoor-training/model/entity/indoor-training.entity';
import { Member } from 'domain-modules/member/model/entity/member.entity';

export class IndoorTrainingSubscriptionUpdatePayload implements Payload {
  @ApiProperty()
  @IsNotEmpty()
  @Length(26, 26)
  indoor_training_subscription_id: UniqueId;
  @ApiProperty()
  @IsNotEmpty()
  purchase_date: Date;
  @ApiProperty({ type: () => IndoorTraining })
  @IsOptional()
  indoor_training: IndoorTraining;
  @ApiProperty({ type: () => Member })
  @IsOptional()
  member: Member;
}

File: capacity_club_mobile_app/lib/domain-features/data/indoor-training-subscription/model/payload/indoor_training_subscription_update_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:capacity_club_mobile_app/domain-features/data/indoor-training/model/indoor_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/member/model/member_model.dart';
import 'package:equatable/equatable.dart';

class IndoorTrainingSubscriptionUpdatePayload extends Equatable
    implements Payload {
  final UniqueId indoor_training_subscription_id;
  final DateTime purchased_date;
  final MemberModel? member;
  final IndoorTrainingModel? indoor_training;

  IndoorTrainingSubscriptionUpdatePayload({
    required this.indoor_training_subscription_id,
    required this.purchased_date,
    this.member,
    required this.indoor_training,
  });

  Map<String, dynamic> toJson() {
    return {
      'indoor_training_subscription_id':
          indoor_training_subscription_id.toJson(),
      'purchased_date': purchased_date.toIso8601String(),
      'member': member ?? {},
      'indoor_training': indoor_training ?? {}
    };
  }

  @override
  List<Object?> get props => [
        indoor_training_subscription_id,
        purchased_date,
        member,
        indoor_training,
      ];
}
